src/core_plugins/kibana/public/management/sections/indices/create_index_pattern/create_index_pattern.js

```diff
@@ -207,39 +195,54 @@ uiModules.get('apps/management')
     return loadingCount > 0;
   };
 
-  this.refreshFieldList = () => {
-    const timeField = this.formValues.timeField;
+  this.refreshTimeFieldOptions = () => {
+    const prevOption = this.formValues.timeFieldOption;
+
     loadingCount += 1;
-    fetchFieldList().then(results => {
-      if (timeField) {
-        updateFieldListAndSetTimeField(results, timeField.name);
-      } else {
-        updateFieldList(results);
-      }
-    }).finally(() => {
-      loadingCount -= 1;
-    });
+    this.timeFieldOptions = [];
+    this.timeFieldOptionsError = null;
+    this.formValues.timeFieldOption = null;
+    getTimeFieldOptions()
```

Comment:
@cjcenizal by that logic there are no function in this PR that qualify unless we reimplement some core services. All of the asynchronicity in this controller is introduced by the `indexPattern.mapper`, `savedObjects`, and `esAdmin` services, all of which use angular async/http libraries.

If you're suggesting that we write more code that is completely disconnected from angular, that only inter-ops with angular at the last minute using something like `$scope.$apply()` or wrapping a non angular promise with `$q.resolve()` then I misunderstood you because I am totally on board with that. Unfortunately we have to adapt a lot of services to not use angular async/http first, or do something (module, app, component) greenfield and use non-angular libraries for everything from async to http.

My stance in this discussion so far was against using angular services in a non-angular context (like an `async function() {}` that uses the `indexPattern.mapper`, or `esAdmin`).

This specific pr is even worse in my opinion, converting any of the helpers to `async function(){}`s would be creating non-angular contexts between angular services and other angular contexts (Angular controller -> non-angular `async function() {}` -> angular services). As you stated, this is something we should probably avoid.